tensorboard/uploader/proto/write_service.proto

```diff
@@ -119,6 +124,125 @@ message WriteScalarResponse {
   // This is empty on purpose.
 }
 
+// The blob sequence writing procedure works as follows:
+//  * The caller requests creation of a new BlobSequence, via the
+//    CreateBlobSequence RPC. The response provides a new BlobSequence ID.
+//  * The caller writes each blob in the sequence separately via the WriteBlob
+//    RPC as described below, providing the blob sequence ID and sequence index
+//    for each blob.
+//  * Multiple WriteBlob streams may operate in parallel, with no constraint
+//    on their ordering.
+
+// Blob sequence uploads are resumable, as follows:
+//  * The client starts the upload process with a new CreateBlobSequence RPC.
+//  * The blob sequences are uniquely keyed by experiment, run, tag, and step.
+//    Since this is a second (or later) attempt regarding an existing sequence,
+//    a new sequence is not created.  Instead the existing sequence ID is
+//    returned.
+//  * The client attempts to upload each blob in the sequence.  These uploads
+//    resume or may be skipped entirely, as needed, according to the below
+//    procedure.
+
+// The blob writing procedure is as follows.  It is intrinsically resumable.
+//  * The caller instantiates a WriteBlob RPC, which is a bidirectional stream.
+//  * The first of these requests must provide write_blob_first_request_data,
```

Comment:
I'm not sure I see the benefit of providing the identifying info for the blob to write on only the first request in the stream.  It's a very small amount of overhead relative to the blob size itself, and I think the client side logic is simpler if it has to maintain less state about which chunk it's on.  In general it seems cleaner if the first chunk sent doesn't look any different from an arbitrary chunk _n_ (except for the final chunk).

The server side already has to keep track of the state, so there's no change there (except it should validate that it's the same blob being written and fail the request if not, but this is nice since it's a cheap check and helps catch client errors and avoids ambiguity).

Making the "key" to WriteBlobRequest mandatory is also just nice structurally since then even in the future if we pass around those request objects (or conceivably want to do something fancy like multiplex several blob writes into a single streaming RPC), we never have to worry about losing track of which blob we're talking about.